Reference/Notifications/Creating-And-Publishing-Notifications.md

```diff
@@ -0,0 +1,115 @@
+---
+versionFrom: 9.0.0
+verified-against: rc002
+meta-title: Creating and Publishing Custom Notifications
+meta.Description: How to create and publish your own custom notifications
+state: complete
+update-links: true
+---
+
+# Creating a custom notification
+
+There may be many reason why you would like to create your own custom notifications, in this article we'll use the CleanUpYourRoom [recurring hosted service](../Scheduling/index-v9.md) as an example, which empties the recycle bin every 5 minutes. You might want to publish a notification once the task has started, and maybe once the task has sucessfully cleared the recycle bin.
+
+For a notification to be publishable there's only one requirement, it must implement the empty marker interface `INotifiaction`, the rest is up to you. For instance we might want to create a notification that just signals that the clean your room task has started and nothing else, in this case we'll create an empty class implementing `INotification`
+
+```C#
+using Umbraco.Cms.Core.Notifications;
+
+namespace Umbraco.Cms.Web.UI.Notifications
+{
+    public class CleanYourRoomStartedNotification : INotification
+    {
+
+    }
+}
+```
+
+This notification can now be published, and we can create a notification handler to recieve it with, see [MediaService-Notifications](MediaService-Notifications.md) for an example of how to implement a notification handler. But this notification alone might not be super helpful, we might want to be able to send some additional information with the notification, however, since this is, in essence, just a normal class, we can include whatever information we want. Let's try and create a `RoomCleanedNotification` which contains the amount of nodes removed from the recycle bin:
```

Comment:
```suggestion
This notification can now be published, and we can create a notification handler to receive it with, see [MediaService-Notifications](MediaService-Notifications.md) for an example of how to implement a notification handler. But this notification alone might not be super helpful, we might want to be able to send some additional information with the notification, however, since this is, in essence, just a normal class, we can include whatever information we want. Let's try and create a `RoomCleanedNotification` which contains the number of nodes removed from the recycle bin:
```